src/module.cc

```diff
@@ -145,22 +144,63 @@ static void OnFatalError(const char* location, const char* message) {
     TriggerNodeReport(Isolate::GetCurrent(), kFatalError, message, location, nullptr);
   }
   fflush(stderr);
-  if (nodereport_core) {
-    raise(SIGABRT); // core dump requested (default)
-  } else {
-    exit(1); // user specified that no core dump is wanted, just exit
-  }
+  raise(SIGABRT);
 }
 
 bool OnUncaughtException(v8::Isolate* isolate) {
-   // Trigger NodeReport if required
+  // Trigger NodeReport if requested
   if (nodereport_events & NR_EXCEPTION) {
     TriggerNodeReport(isolate, kException, "exception", __func__, nullptr);
-  } 
-  return nodereport_core;
+  }
+  if ((commandline_string.find("abort-on-uncaught-exception") != std::string::npos) ||
+      (commandline_string.find("abort_on_uncaught_exception") != std::string::npos)) {
+    return true;  // abort required
+  }
+  // On Node v4 and v6 we need to print the exception backtrace to stderr, as
+  // Node does not do so (unless we trigger an abort, as above)
+  if ((version_string.find("v4") != std::string::npos) ||
+      (version_string.find("v6") != std::string::npos)) {
+    fprintf(stderr, "\nUncaught exception at:\n");
+#ifdef _WIN32
+    // On Windows, print the stack using StackTrace API
+    PrintStackFromStackTrace(isolate, stderr);
+#else
+    // On other platforms use the Message API
+    Message::PrintCurrentStackTrace(isolate, stderr);
+#endif
+  }
+  return false;
 }
 
-#ifndef _WIN32
+#ifdef _WIN32
+static void PrintStackFromStackTrace(Isolate* isolate, FILE* fp) {
+  Local<StackTrace> stack = StackTrace::CurrentStackTrace(isolate, 255,
+                                                          StackTrace::kDetailed);
+  // Print the JavaScript function name and source information for each frame
+  for (int i = 0; i < stack->GetFrameCount(); i++) {
+    Local<StackFrame> frame = stack->GetFrame(i);
+    Nan::Utf8String fn_name_s(frame->GetFunctionName());
+    Nan::Utf8String script_name(frame->GetScriptName());
+    const int line_number = frame->GetLineNumber();
+    const int column = frame->GetColumn();
+
+    if (frame->IsEval()) {
+      if (frame->GetScriptId() == Message::kNoScriptIdInfo) {
+        fprintf(fp, "at [eval]:%i:%i\n", line_number, column);
+      } else {
+        fprintf(fp, "at [eval] (%s:%i:%i)\n", *script_name, line_number, column);
+      }
+    } else {
+      if (fn_name_s.length() == 0) {
+        fprintf(fp, "%s:%i:%i\n", *script_name, line_number, column);
+      } else {
+        fprintf(fp, "%s (%s:%i:%i)\n", *fn_name_s, *script_name, line_number, column);
+      }
```

Comment:
As above, the stack trace in a NodeReport file identifies constructors, but that is not needed in the simpler version we are outputting to stderr.